### DISCUSSION TITLE
regarding github pages

### DISCUSSION THREAD

**Discussion by Harish-2003** [community/community#126734] [New to GitHub]
hey i created an repo and i want to host it on github pages without the creation of myusername.github.io can i do that is it possible??

**Comment by pedrolemos4** [comment]
Yes, it is detailed in the tutorial: https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site

**Comment by davevad93** [comment]
Hi @Harish-2003 ,

If you mean using a custom domain instead of the default `username.github.io`, there's a detailed guide in the [docs](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site) on how to do it.

1. [About custom domains and GitHub Pages](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages)
2. [Managing a custom domain for your GitHub Pages site](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site)
3. [Verifying your custom domain for GitHub Pages](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/verifying-your-custom-domain-for-github-pages)
4. [Troubleshooting custom domains and GitHub Pages](https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/troubleshooting-custom-domains-and-github-pages)

**Reply by Harish-2003** [reply]
Can I use something like reponame.github.io

**Reply by davevad93** [reply]
Hey @Harish-2003  I don't think it's possible, as far as I know the only possible options are `username.github.io/project` or `orgname.github.io/project`, otherwise the only alternative would be to create a custom domain.

**Reply by Harish-2003** [reply]
Do I need to own a domain?

**Reply by davevad93** [reply]
Yes, you have to own and register your custom domain name.

**Comment by namdng09** [comment]
Hi there! i found solution of other people, i hope this comment can answer your question.

If you want to create an orginization/user site just create a repository named yourUserName.github.io. Then, follow the tutorials on how to set up a GH Pages site in that repository.

If you want to create a site for a single repository, just create a branch named “gh-pages” and put your site in there, according to the tutorials.

If you want to access your user/orginazation site just head over to yourUserName.github.io. If you want to access a repository site just head over to yourUserName.github.io/respositoryName!

The following link acutally contains all info on what I just said: https://help.github.com/articles/user-organization-and-project-pages/.

Thankyou for reading and have a good day <3

**Reply by Harish-2003** [reply]
I don't want to use my my username.github.io I want to use the repo name.github.io is it possible
